rest/route/github_routes.go

```diff
@@ -0,0 +1,141 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/patch"
+	"github.com/evergreen-ci/evergreen/rest"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/google/go-github/github"
+	"github.com/mongodb/grip"
+)
+
+type githubHookApi struct {
+	secret []byte
+
+	event interface{}
+	msgId string
+}
+
+func getGithubHooksRouteManager(route string, version int) *RouteManager {
+	methods := []MethodHandler{}
+	if secret := getWebhookSecret(); len(secret) > 0 {
+		methods = append(methods, MethodHandler{
+			Authenticator: &NoAuthAuthenticator{},
+			RequestHandler: &githubHookApi{
+				secret: secret,
+			},
+			MethodType: http.MethodPost,
+		})
+
+	} else {
+		grip.Warning("Github webhook secret is empty! Github webhooks have been disabled!")
+	}
+
+	return &RouteManager{
+		Route:   route,
+		Methods: methods,
+		Version: version,
+	}
+}
+
+func (gh *githubHookApi) Handler() RequestHandler {
+	return &githubHookApi{
+		secret: getWebhookSecret(),
+	}
+}
+
+func getWebhookSecret() []byte {
+	secret := ""
+	if settings := evergreen.GetEnvironment().Settings(); settings != nil {
```

Comment:
let's find a way to get this without this operation, and instead passing the secret into the constructor on line (which will involve a bit of surgery deeper down, but I think is cleaner in the end.) 

We also spent some time trying to unwind from this in the spawn a host rest endpoint, which we resolved by implementing the route using the legacy api (which I think is viable here too.)